lib/matplotlib/axes/_axes.py

```diff
@@ -4224,29 +4224,33 @@ def _parse_scatter_color_args(c, edgecolors, kwargs, xsize,
             c = (facecolors if facecolors is not None
                  else "b" if rcParams['_internal.classic_mode']
                  else get_next_color_func())
-
-        # After this block, c_array will be None unless
-        # c is an array for mapping.  The potential ambiguity
-        # with a sequence of 3 or 4 numbers is resolved in
-        # favor of mapping, not rgb or rgba.
-        # Convenience vars to track shape mismatch *and* conversion failures.
+        c_is_string_or_strings = (
+            isinstance(c, str)
+            or (isinstance(c, collections.abc.Iterable) and len(c) > 0
+                and isinstance(cbook.safe_first_element(c), str)))
+
+        # After this block, c_array will be None unless c is an array for
```

Comment:
Everytime I get back here, I'm wondering what `c_array` is for. It's basically a local variable in the try/else block. As stated here, the checking `c_array is None` further below carries some additional information. Wouldn't it be more clear to use a boolean like `c_is_explicit_color` or `c_should_be_mapped` here?

Or alternatively call this array `c_array_for_mapping` at least?

Reply:
Indeed, moving that info in a boolean variable greatly improves readability.  Pushed a separate commit with that.